cumulusci/core/config/OrgConfig.py

```diff
@@ -63,6 +67,26 @@ def _refresh_token(self, keychain, connected_app):
     def lightning_base_url(self):
         return self.instance_url.split(".")[0] + ".lightning.force.com"
 
+    @property
+    def salesforce_client(self):
+        if not self._client:
+            self._client = Salesforce(
+                instance=self.instance_url.replace("https://", ""),
+                session_id=self.access_token,
+                version=45,
+            )
+        return self._client
+
+    @property
+    def latest_api_version(self):
+        if not self._latest_api_version:
+            response = self.salesforce_client._call_salesforce(
+                "GET",
+                "https://{}/services/data".format(self.salesforce_client.sf_instance),
+            )
```

Comment:
Use `client.restful("services/data", method="GET")` to avoid using an internal method and to take advantage of the fact that the client already knows how to construct the full URL.

Reply:
```
>>> from cumulusci.salesforce_api.utils import get_simple_salesforce_connection
>>> from cumulusci.cli import cci
>>> config = cci.load_config()
>>> org_name, org_config = config.get_org("dev")
>>> sf = get_simple_salesforce_connection(config.project_config, org_config)
2019-10-29 16:39:38: Getting scratch org info from Salesforce DX
sf.restful("services/data")
>>> sf.restful("services/data")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/pprescod/code/envs/cumulusci-py3/lib/python3.7/site-packages/simple_salesforce/api.py", line 305, in restful
    **kwargs)
  File "/Users/pprescod/code/envs/cumulusci-py3/lib/python3.7/site-packages/simple_salesforce/api.py", line 478, in _call_salesforce
    exception_handler(result, name=name)
  File "/Users/pprescod/code/envs/cumulusci-py3/lib/python3.7/site-packages/simple_salesforce/util.py", line 61, in exception_handler
    raise exc_cls(result.url, result.status_code, name, response_content)
simple_salesforce.exceptions.SalesforceResourceNotFound: Resource services/data Not Found. Response content: [{'errorCode': 'NOT_FOUND', 'message': 'The requested resource does not exist'}]

```

With debug logging turned on I can see:

DEBUG:urllib3.connectionpool:https://flow-page-8225-dev-ed.cs40.my.salesforce.com:443 "GET /services/data/v46.0/services/data HTTP/1.1" 404 None

The same thing happens if I add a leading slash to the relative URL. With enough ../../.. stuff I can get it to work but that presumably is not what you were intending.

Reply:
@prescod sorry, I forgot it adds that portion of the path too. We can keep using _call_salesforce here.